docs/src/main/asciidoc/deploying-to-google-cloud.adoc

```diff
@@ -0,0 +1,236 @@
+- Build your application as a native executable using `mvn clean package -Pnative`.
+
+Finally, launch `gcloud app deploy` on the root of your project, it will upload all your project files (the list can be reduced via the  `.gcloudignore` file),
+build your Dockerfile and launch it on App Engine Flexible custom runtime.
+
+It uses Cloud Build to build your Docker image and deploy it to Google Container Registry (GCR).
+
+When done, the output will display the URL of your application (target url), you can use it with curl or directly open it on your browser using `gcloud app browse`.
+
+As the getting started endpoint is located on the `/hello` path, make sure to navigate to it.
+
+NOTE: App Engine Flexible custom runtimes support link:https://cloud.google.com/appengine/docs/flexible/custom-runtimes/configuring-your-app-with-app-yaml#updated_health_checks[health checks],
+it is strongly advised to provide them thanks to Quarkus link:microprofile-health[Microprofile Heath] support.
+
+== Deploying to Google Cloud Run
+
+Google Cloud Run allows you to run your Docker containers inside Google Cloud Platform in a managed way.
+
+NOTE: By default, Quarkus listens on port 8080, and it's also the Cloud Run default port.
+No need to use the `PORT` environment variable defined in Cloud Run to customize the Quarkus HTTP port.
+
+Cloud Run will use Cloud Build to build your Docker image and deploy it to Google Container Registry (GCR).
+
+Both JVM and native executable versions will work.
+
+To deploy a JVM application:
+
+- Copy the JVM Dockerfile to the root directory of your project: `cp src/main/docker/Dockerfile.jvm Dockerfile`.
+- Build your application using `mvn clean package`.
+
+To deploy a native application:
+
+- Copy the JVM Dockerfile to the root directory of your project: `cp src/main/docker/Dockerfile.native Dockerfile`.
+- Build your application as a native executable using `mvn clean package -Pnative`.
+
+Then, create a `.gcloudignore` file to tell gcloud which file should be not be uploaded for Cloud Build,
+without it, it defaults to `.gitignore` that usually exclude the target directory where you packaged application has been created.
+
+In this example, I only exclude the `src` directory:
+
+[source]
+----
+src/
+----
+[source]
+
+Then, use Cloud Build to build your image, it will upload to a Google Cloud Storage bucket all the files of your application (except the ones ignored by the `.gcloudignore`file),
+build your Docker image and push it to Google Container Registry (GCR).
+
+[source, shell script]
+----
+gcloud builds submit --tag gcr.io/PROJECT-ID/helloworld
+----
+
+NOTE: You can also build your image locally and push it to a publicly accessible Docker registry, then use this image in the next step.
+
+Finally, use Cloud Run to launch your application.
+
+[source, shell script]
+----
+gcloud run deploy --image gcr.io/PROJECT-ID/helloworld --platform managed
+----
```

Comment:
I haven't tried it for some time, but are native images working at CloudRun without setting `DISABLE_SIGNAL_HANDLERS` now?

Oh... I see it was you who reported it works: https://github.com/quarkusio/quarkus/issues/4262

Will try it by myself next week, thank you.

Reply:
Yes, it works without it now.